crates/trippy-core/src/tos.rs

```diff
@@ -0,0 +1,206 @@
+use crate::{Error, TypeOfService};
+use std::fmt::Display;
+
+/// Explicit Congestion Notification (`ECN`).
+///
+/// This is used in the `ECN` field of the `IP` header.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum ExplicitCongestionNotification {
+    /// Not ECN-Capable Transport (00, 0 dec)
+    NotECT,
+    /// ECN Capable Transport(1) (01, 1 dec)
+    ECT1,
+    /// ECN Capable Transport(0) (10, 2 dec)
+    ECT0,
+    /// Congestion Experienced (11, 3 dec)
+    CE,
+}
+
+impl TryFrom<u8> for ExplicitCongestionNotification {
+    type Error = Error;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::NotECT),
+            1 => Ok(Self::ECT1),
+            2 => Ok(Self::ECT0),
+            3 => Ok(Self::CE),
+            _ => Err(Error::Other(format!("invalid ECN value: {value}"))),
+        }
+    }
+}
```

Comment:
Probably a silly question, why not define the decimal values in the enumerator (assuming that this is possible in rust) then the name value association would exist in one pace only?

Reply:
There are some Rust proc macro crates to do this, I prefer to keep it explicit in Trippy (less "magic")